modules_basic/message.js

```diff
@@ -1,122 +1,86 @@
+    if (content) return mini(msg, content)
 
-    var el = api.message_content(msg)
-    if(!el) return mini(msg, message_content_mini_fallback(msg))
+    var content = api.message_content(msg)
+    if (!content) return mini(msg, message_content_mini_fallback(msg))
 
-    var links = []
-    for(var k in CACHE) {
-      var _msg = CACHE[k]
-      if(Array.isArray(_msg.content.mentions)) {
-        for(var i = 0; i < _msg.content.mentions.length; i++)
-          if(_msg.content.mentions[i].link == msg.key)
-          links.push(k)
+    return h('Message', {
+      'ev-keydown': navigateToMessageOnEnter,
+      attributes: {
+        tabindex: '0'
+      }
+    }, [
+      h('header.author', api.message_author(msg)),
+      h('section.meta', api.message_meta(msg)),
+      h('section.content', content),
+      h('section.action', api.message_action(msg)),
+      h('footer.backlinks', api.message_backlinks(msg))
+    ])
+
+    function navigateToMessageOnEnter (ev) {
+      // on enter, hit first meta.
+      if(ev.keyCode == 13) {
+
+        // unless in an input
+        if (ev.target.nodeName === 'INPUT'
+          || ev.target.nodeName === 'TEXTAREA') return
+
+        // HACK! (mw)
+        // there's no exported api to open a new tab. :/
+        // it's only done in `app.js` module in an`onhashchange` handler.
+        // sooooooo yeah this shit for now :)
+        var wtf = h('a', { href: `#${msg.key}` })
+        msgEl.appendChild(wtf)
+        wtf.click()
       }
     }
+  }
 
-    var backlinks = h('div.backlinks')
-    if(links.length)
-      backlinks.appendChild(h('label', 'backlinks:', 
-        h('div', links.map(function (key) {
-          return api.message_link(key)
-        }))
-      ))
-
-
-  //  pull(
-  //    sbot_links({dest: msg.key, rel: 'mentions', keys: true}),
-  //    pull.collect(function (err, links) {
-  //      if(links.length)
-  //        backlinks.appendChild(h('label', 'backlinks:', 
-  //          h('div', links.map(function (link) {
-  //            return message_link(link.key)
-  //          }))
-  //        ))
-  //    })
-  //  )
-
-    var msg = h('div.message',
-      h('div.title.row',
-        h('div.avatar', api.avatar(msg.value.author, 'thumbnail')),
-        h('div.message_meta.row', api.message_meta(msg))
-      ),
-      h('div.message_content', el),
-      h('div.message_actions.row',
-        h('div.actions', api.message_action(msg),
-          h('a', {href: '#' + msg.key}, 'Reply')
-        )
-      ),
-      backlinks,
-      {onkeydown: function (ev) {
-        //on enter, hit first meta.
-        if(ev.keyCode == 13) {
-
-          // unless in an input
-          if (ev.target.nodeName === 'INPUT'
-            || ev.target.nodeName === 'TEXTAREA') return
-
-          msg.querySelector('.enter').click()
-        }
-      }}
-    )
-
-    // ); hyperscript does not seem to set attributes correctly.
-    msg.setAttribute('tabindex', '0')
-
-    return msg
+  function mini(msg, el) {
```

Comment:
this pattern feels a bit weird. I had to read it a couple of times, this feels like a template.
I wonder if it would be good to have a `miniTemplate` and a `standardTemplate` both extracted so that we could see this pattern more clearly?